src/us/kbase/workspace/test/controllers/sample/SampleServiceController.java

```diff
@@ -0,0 +1,229 @@
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.ini4j.Ini;
+import org.ini4j.Profile.Section;
+
+import us.kbase.workspace.test.controllers.arango.ArangoController;
+
+/** Q&D Utility to run the Samnple Service for the purposes of testing from Java.
+ * @author gaprice@lbl.gov
+ *
+ */
+public class SampleServiceController {
+	
+	private static final String SAMPLE_SERVICE_NAME = "SampleService";
+	
+	private final Process sampleService;
+	private final int port;
+	private final Path tempDir;
+	private final Path logfile;
+	
+	/** A container for a multitude of arangoDB parameters required by the Sample Service
+	 * controller.
+	 *
+	 */
+	public static class SampleServiceArangoParameters {
+		// ugh.
+		public final String arangoUser;
+		public final String arangoPwd;
+		public final String arangoDB;
+		public final String sampleCollection;
+		public final String versionCollection;
+		public final String versionEdgeCollection;
+		public final String nodeCollection;
+		public final String nodeEdgeCollection;
+		public final String dataLinkCollection;
+		public final String workspaceObjectVersionShadowCollection;
+		public final String schemaCollection;
+		
+		/** Create the parameter container.
+		 * @param arangoUser the ArangoDB user account to use to connect to ArangoDB.
+		 * @param arangoPwd the user's password.
+		 * @param arangoDB the database to connect to.
+		 * @param sampleCollection the collection in which to store sample information.
+		 * @param versionCollection the collection in which to store version information.
+		 * @param versionEdgeCollection the collection in which to store version edge information.
+		 * @param nodeCollection the collection in which to store node information.
+		 * @param nodeEdgeCollection the collection in which to store node edge information.
+		 * @param dataLinkCollection the collection in which to store data link information.
+		 * @param workspaceObjectVersionShadowCollection the collection in which to store
+		 * workspace shadow object information.
+		 * @param schemaCollection the collection in which to store schema information.
+		 */
+		public SampleServiceArangoParameters(
+				final String arangoUser,
+				final String arangoPwd,
+				final String arangoDB,
+				final String sampleCollection,
+				final String versionCollection,
+				final String versionEdgeCollection,
+				final String nodeCollection,
+				final String nodeEdgeCollection,
+				final String dataLinkCollection,
+				final String workspaceObjectVersionShadowCollection,
+				final String schemaCollection) {
+			this.arangoUser = checkString(arangoUser, "arangoUser");
+			this.arangoPwd = checkString(arangoPwd, "arangoPwd");
+			this.arangoDB = checkString(arangoDB, "arangoDB");
+			this.sampleCollection = checkString(sampleCollection, "ampleCollection");
```

Comment:
string should be "sampleCollection" ?

Reply:
Fixed this locally, will be in a future commit.

Reply:
Why not this PR?

Reply:
Because locally there's 21 commits sitting on top of this PR and I don't want to go through the annoyance of making a new branch, fixing the message, and then cherry picking all the other commits on top of that branch for such a trivial change

Reply:
If it actually affected functionality then sure